controllers/flinkcluster_observer.go

```diff
@@ -263,6 +268,51 @@ func (observer *ClusterStateObserver) observeFlinkJobs(
 	}
 }
 
+func (observer *ClusterStateObserver) observeSavepoint(observed *ObservedClusterState) error {
+	var log = observer.log
+
+	if observed.cluster == nil {
+		return nil
+	}
+
+	// Get savepoint status in progress.
+	var savepointStatus = observed.cluster.Status.Savepoint
+	if savepointStatus != nil &&
+		savepointStatus.State == SavepointStateProgressing &&
+		savepointStatus.TriggerID != "" {
+		var jobID = savepointStatus.JobID
+		var triggerID = savepointStatus.TriggerID
+		var savepoint flinkclient.SavepointStatus
+		var flinkAPIBaseURL = getFlinkAPIBaseURL(observed.cluster)
+
+		var retryCount = 0
+		var retriable = func(err error) bool {
+			if httpErr, ok := err.(*flinkclient.HTTPError); ok && httpErr.StatusCode >= 500 {
+				log.Info("Savepoint status request is failed", "http error", httpErr, "jobID", jobID, "triggerID", triggerID, "retry", retryCount)
+				retryCount++
```

Comment:
Did you consider avoiding retry here and simply rely on the reconciliation loop to retry the observation in the next iteration? That could reduce the complexity.

Reply:
As for the savepoint status check, it will be checked continuously in the reconcile loop, so it seems good to delete the backoff retry in the observer.

Reply:
Fixed